### PR TITLE
Feature/50 new database

### PR DESCRIPTION
**Pull request by mtl3jx**

### CODE REVIEW

File: app/src/main/java/org/septa/android/app/MainActivity.java [hunk 1]
```diff
@@ -365,6 +418,140 @@ public void goToSchedulesForTarget(StopModel start, StopModel destination, Trans
         switchToSchedules(bundle);
     }
 
+    @Override
```
Comment (josephkampf) [inline app/src/main/java/org/septa/android/app/MainActivity.java]: There is a lot of logic in this MainActvity around the database expansion.  Encapsulate this logic in its own class.

File: app/src/main/java/org/septa/android/app/database/CheckForLatestDB.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    public CheckForLatestDB(CheckForLatestDBListener listener) {
+        this.mListener = listener;
+    }
+
+    @Override
+    protected Void doInBackground(Object... voids) {
+        try {
+            // connect to URL
+            URL jsonDataUrl = new URL(SEPTADatabase.getLatestDatabaseApiUrl());
```
Comment (josephkampf) [inline app/src/main/java/org/septa/android/app/database/CheckForLatestDB.java]: So you are doing a lot of manual IO stream stuff here.  You should follow the same pattern as all the other APIs.

File: app/src/main/java/org/septa/android/app/database/CheckForLatestDB.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+                stringBuffer.append(line);
+            }
+
+            // grab latest DB metadata
+            JSONObject latestDBMetadata = new JSONObject(stringBuffer.toString());
+            latestDBVersion = latestDBMetadata.getInt("version");
+            latestDBURL = latestDBMetadata.getString("url");
+            updatedDate = latestDBMetadata.getString("updateDate");
+        } catch (MalformedURLException e) {
+            e.printStackTrace();
```
Comment (josephkampf) [inline app/src/main/java/org/septa/android/app/database/CheckForLatestDB.java]: We should be logging these errors.  Printing the stack trace only works for LogCat.

File: app/src/main/java/org/septa/android/app/database/SEPTADatabase.java [hunk 4]
```diff
@@ -14,37 +14,73 @@
     /**
      * Current packaged DB version, update number when packaged DB changes
      */
-    private static final int DATABASE_VERSION = 282;
-    private static final String DATABASE_FILE_NAME = "SEPTA.sqlite";
+    private static final String LATEST_DATABASE_API_URL = "https://s3.amazonaws.com/mobiledb.septa.org/latest/latestDb.json";
```
Comment (josephkampf) [inline app/src/main/java/org/septa/android/app/database/SEPTADatabase.java]: Externalize this String into Values.

File: app/src/main/java/org/septa/android/app/database/SEPTADatabase.java [hunk 5]
```diff
@@ -14,37 +14,73 @@
...
+
+    // these are left in case the user does not have the most up to date version of the database
+    // modify databaseVersion and databaseFileName when pushing out a new release with a new DB
+    private static int databaseVersion = 282;
```
Comment (josephkampf) [inline app/src/main/java/org/septa/android/app/database/SEPTADatabase.java]: So shouldn't the DB that is shipped with the App follow the same naming convention as the downloaded one?  This way you can keep everything the same?  We should modify the gradle build to download the latest version of the database along with its version number and put the db version into the app.

File: app/src/main/java/org/septa/android/app/database/SEPTADatabase.java [hunk 6]
```diff
@@ -14,37 +14,73 @@
...
-//                "insert into stop_route_direction(stop_id, route_id, direction_id) SELECT DISTINCT S.stop_id, T.route_id, direction_id   FROM trips_bus T join stop_times_bus ST ON T.trip_id = ST.trip_id join stops_bus S on ST.stop_id = S.stop_id",
-//                "insert into stop_route_direction(stop_id, route_id, direction_id) SELECT DISTINCT S.stop_id, T.route_id, direction_id   FROM trips_MFL T join stop_times_MFL ST ON T.trip_id = ST.trip_id join stops_bus S on ST.stop_id = S.stop_id",
-//                "insert into stop_route_direction(stop_id, route_id, direction_id) SELECT DISTINCT S.stop_id, T.route_id, direction_id   FROM trips_BSL T join stop_times_BSL ST ON T.trip_id = ST.trip_id join stops_bus S on ST.stop_id = S.stop_id",
-//                "CREATE INDEX stop_srd_index ON stop_route_direction (stop_id)",
-//                "CREATE INDEX route_srd_index ON stop_route_direction (route_id)",
-//                "CREATE INDEX direction_srd_index ON stop_route_direction (direction_id)"});
-
+        // create indices on new app database
+        execSQL(new String[]{
+                "CREATE INDEX stop_srd_index ON stop_route_direction (stop_id);",
```
Comment (josephkampf) [inline app/src/main/java/org/septa/android/app/database/SEPTADatabase.java]: These should be externalized.  You can store them in a text file that you read in or you can put them into the Strings values with the other SQL.  But hard coding them is not a good idea.

Comment (josephkampf) [reply app/src/main/java/org/septa/android/app/database/SEPTADatabase.java]: You can also execute them all in 1 shot if they are separated by ";"

File: app/src/main/java/org/septa/android/app/database/SEPTADatabase.java [hunk 7]
```diff
@@ -14,37 +14,73 @@
...
     }
 
     private void execSQL(String statements[]) {
         try {
-            for (String sql : statements)
+            for (String sql : statements) {
                 getWritableDatabase().execSQL(sql);
+            }
         } catch (SQLException e) {
             //e.printStackTrace();
```
Comment (josephkampf) [inline app/src/main/java/org/septa/android/app/database/SEPTADatabase.java]: Should log this error.

File: app/src/main/java/org/septa/android/app/database/SEPTADatabase.java (hunk 4 above)
Comment (mtl3jx) [reply app/src/main/java/org/septa/android/app/database/SEPTADatabase.java]: I have been putting strings that are displayed to the user into string resources. (Supposing the app had multiple languages, localization would be useful here.) Strings only used internally or for the code are better kept as string constants.

File: app/src/main/java/org/septa/android/app/database/CheckForLatestDB.java (hunk 2 above)
Comment (mtl3jx) [reply app/src/main/java/org/septa/android/app/database/CheckForLatestDB.java]: resolved

File: app/src/main/java/org/septa/android/app/database/SEPTADatabase.java (hunk 5 above)
Comment (mtl3jx) [reply app/src/main/java/org/septa/android/app/database/SEPTADatabase.java]: done

File: app/src/main/java/org/septa/android/app/MainActivity.java (hunk 1 above)
Comment (mtl3jx) [reply app/src/main/java/org/septa/android/app/MainActivity.java]: done

**Pull request merged**